File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        m=len(matrix)
        n=m

        #transpose nikalna hai 
        for i in range(0,m):
            for j in range(i,n):
                #diagonal se swap krna hai 
                matrix[i][j],matrix[j][i]=matrix[j][i],matrix[i][j]

        
        #reverse krna hai 
        for row in matrix:
            row.reverse()

File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        //n*n
        int m=matrix.size();
        int n=m;

        //transpose nikalo
        for(int i=0;i<m;i++){
            for(int j=i;j<n;j++){
                //diagonal element se swap kr do ... upper wale se niche walo ko ... col ko row bna do
                swap(matrix[i][j],matrix[j][i]);

            }
        }
        //reverse kr do transpose matrix ko
        for(int i=0;i<m;i++){
            reverse(matrix[i].begin(),matrix[i].end());
        } 

        
    }
};